launcher/src/main/java/org/apache/spark/launcher/SparkLauncher.java

```diff
@@ -73,6 +74,21 @@
   /** Logger name to use when launching a child process. */
   public static final String CHILD_PROCESS_LOGGER_NAME = "spark.launcher.childProcLoggerName";
 
+  /** Launcher Server Port to use when launching a child process. */
+  public static final String LAUNCHER_INTERNAL_PORT = "spark.launcher.internal.port";
+
+  /** Launcher Server sets this when launching a child process. */
+  public static final String CHILD_PROCESS_LAUNCHER_INTERNAL_SECRET = "spark.launcher.internal.secret";
+
+  /** Stop Flag if interrupted Launcher Server goes away. */
+  public static final String CHILD_PROCESS_LAUNCHER_STOP_FLAG = "spark.launcher.internal.stop.flag";
+
+  public SparkLauncher stopIfInterrupted() {
+    this.stopIfInterrupted = true;
+    return this;
+  }
+
+  private boolean stopIfInterrupted = false;
```

Comment:
can you move this up above the function so its at the top of the class by other variable defs